python/tank/authentication/shotgun_shared/saml2_sso.py

```diff
@@ -582,17 +611,163 @@ def on_sso_disable_renewal(self, event):
+
+    :returns: A string user id value, or None.
+    """
+    user_id = None
+    user_domain = None
+    for cookie in cookies:
+        # Shotgun appends the unique numerical ID of the user to the cookie name:
+        # ex: shotgun_sso_session_userid_u78
+        if cookie.startswith("shotgun_sso_session_userid_u"):
+            if user_id is not None:
+                # Should we find multiple cookies with the same prefix, it means
+                # that we are using cookies from a multi-session environment. We
+                # have no way to identify the proper user id in the lot.
+                message = "The cookies for this user seem to come from two differen shotgun side: '%s' and '%s'"
+                raise Saml2SssoMultiSessionNotSupportedError(message % (user_domain, cookies[cookie]['domain']))
+            user_id = cookie[28:]
+            user_domain = cookies[cookie]['domain']
+    return user_id
+
+
+def _get_cookie_from_prefix(encoded_cookies, cookie_prefix):
+    """
+    Returns a cookie value based on a prefix to which we will append the user id.
+
+    :param encoded_cookies: An encoded string representing the cookie jar.
+    :param cookie_prefix: The prefix of the cookie name.
+
+    :returns: A string of the cookie value, or None.
+    """
+    value = None
+    cookies = _decode_cookies(encoded_cookies)
+    key = "%s%s" % (cookie_prefix, _get_shotgun_user_id(cookies))
+    if key in cookies:
+        value = cookies[key].value
+    return value
+
+
+def is_sso_enabled(encoded_cookies):
+    """
+    Indicate if SSO is being used from the Shotgun cookies.
+
+    :param encoded_cookies: An encoded string representing the cookie jar.
+
+    :returns: True or False
+    """
+    cookies = _decode_cookies(encoded_cookies)
+    return _get_shotgun_user_id(cookies) is not None
+
+
+def get_saml_claims_expiration(encoded_cookies):
+    """
+    Obtain the expiration time of the saml claims from the Shotgun cookies.
+
+    :param encoded_cookies: An encoded string representing the cookie jar.
+
+    :returns: An int with the time in seconds since January 1st 1970 UTC, or None
+    """
+    # Shotgun appends the unique numerical ID of the user to the cookie name:
+    # ex: shotgun_sso_session_expiration_u78
+    saml_claims_expiration = _get_cookie_from_prefix(encoded_cookies, "shotgun_sso_session_expiration_u")
+    if saml_claims_expiration is not None:
+        saml_claims_expiration = int(saml_claims_expiration)
+    return saml_claims_expiration
+
+
+def get_saml_user_name(encoded_cookies):
+    """
+    Obtain the saml user name from the Shotgun cookies.
+
+    :param encoded_cookies: An encoded string representing the cookie jar.
+
+    :returns: A string with the user name, or None
+    """
+    # Shotgun appends the unique numerical ID of the user to the cookie name:
+    # ex: shotgun_sso_session_userid_u78
+    user_name = _get_cookie_from_prefix(encoded_cookies, "shotgun_sso_session_userid_u")
```

Comment:
maybe this is part of your upcoming refactor, but at some point please turn the repeated `"shotgun_sso_session_userid_u"` into a constant :)